Test against Ruby 2.2 and use bundler cache

dlackty opened the pull request.

tomykaira commented:
Sorry, but I do not have time to fix the failing tests.
I'll merge if CI passes.

dlackty commented:
@tomykaira Fixed. 

As [test-unit is no longer included in Ruby 2.2](https://bugs.ruby-lang.org/issues/9711), I added it as `development_dependency`. 

`contest` gem was also removed due to the incompatibility for new version of `test-unit`, but fortunately `ActiveSupport` provides similar functionalities.

tomykaira commented:
> fortunately ActiveSupport provides similar functionalities.

I didn't know that. :astonished: 
`contest` is exactly why I hesitated to pick latest `test-unit` or `minitest`, so I cannot thank you enough!
Thanks again for quick and elegant work.
